### ISSUE TITLE
Clear Datasource fails with 500 Internal Server Error

### ISSUE THREAD

**Issue by johnjohndoe** [52North/SOS#288]
I am trying to delete all previously imported data. Therefore I visit *Admin / Datasource Maintenance* under `http://localhost:8080/52n-sos-webapp/admin/datasource`. I click at <kbd>Clear Datasource</kbd>. This **fails** with the following error:

> Error! Request failed: 500 Internal Server Error

![Clear Datasource](https://cloud.githubusercontent.com/assets/144518/8997582/4de95c12-3723-11e5-808a-a0baf2c2c0a3.png)

The server log in `/var/log/tomcat7/52n-sos-webapp.log` reveals nothing helpful:

```
INFO [http-bio-8080-exec-25] [AdminDatasourceController.java:125] Clearing database contents.
INFO [http-bio-8080-exec-25] [Configuration.java:2073] HHH000043: Configuring from resource: /sos-hibernate.cfg.xml
INFO [http-bio-8080-exec-25] [Configuration.java:2092] HHH000040: Configuration resource: /sos-hibernate.cfg.xml
INFO [http-bio-8080-exec-25] [Configuration.java:2214] HHH000041: Configured SessionFactory: null
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/Offering.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/FeatureOfInterestType.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/Procedure.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/ObservableProperty.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/FeatureOfInterest.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/ObservationType.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/Unit.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/Codespace.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/ProcedureDescriptionFormat.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/core/ObservationConstellation.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/series/observation/SeriesValueTime.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/series/observation/Series.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/series/observation/SeriesValue.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/series/observation/SeriesObservation.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/series/observation/SeriesObservationInfo.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/series/observation/SeriesObservationTime.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/transactional/RelatedFeatureRole.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/transactional/ValidProcedureTime.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/transactional/RelatedFeature.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/transactional/TObservableProperty.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/transactional/ResultTemplate.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/transactional/TProcedure.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/transactional/TFeatureOfInterest.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/transactional/TOffering.hbm.xml
INFO [http-bio-8080-exec-25] [Configuration.java:498] HHH000220: Reading mappings from file: /var/lib/tomcat7/webapps/52n-sos-webapp##4.3.0-webapp-bundle/WEB-INF/classes/mapping/transactional/Parameter.hbm.xml
```

### Workaround

@CarstenHollmann told me that the [`PG_series_clear_database.sql`](https://github.com/52North/SOS/blob/4.3.0/misc/db/PostgreSQL/series/PG_series_clear_database.sql) script can be used instead. Make sure to remove the `i18n*` tables if you did not configure internationalization while setting up SOS. Further, you might have to change the schema from `public` to whatever you defined.

**Comment by CarstenHollmann** [comment]
I guess you use a different scheme than the default schemas (public).

Currently the SOS does not consider the defined schema when creating the clear database table statement.

**Comment by johnjohndoe** [comment]
Yes, I am using another schema.

**Comment by CarstenHollmann** [comment]
Thanks.
We will implement the consideration of the defined scheme for **Clear dDatasource** in future versions.

**Comment by CarstenHollmann** [comment]
Fixed in #290

**Comment by johnjohndoe** [comment]
:+1:
